middlewares/metrics.go

```diff
@@ -1,12 +1,35 @@
 package middlewares
 
 import (
+	"fmt"
 	"net/http"
+	"os"
 	"time"
 
+	"github.com/DataDog/datadog-go/statsd"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+var statsdClient *statsd.Client
+
+func initStatsdClient() {
+	statsdHost := os.Getenv("STATSD_HOST")
```

Comment:
If you wanted to make this lib more flexible, accept these as arguments to the function. That way the caller can get them as either env vars or flags or whatever.